internal/scylla/client.go

```diff
@@ -126,88 +122,56 @@ func (c *Client) newHttpRequest(ctx context.Context, method, path string, reqBod
 	return req, nil
 }
 
-func (c *Client) doHttpRequest(req *http.Request) (resp *http.Response, temporaryErr bool, err error) {
-	resp, err = c.HTTPClient.Do(req)
-	if err != nil {
-		if oe, ok := err.(*net.OpError); ok {
-			temporaryErr = oe.Temporary()
-		}
-
-		return
-	}
-
-	temporaryErr = resp.StatusCode == http.StatusBadGateway || resp.StatusCode == http.StatusGatewayTimeout || resp.StatusCode == http.StatusTooManyRequests
-	return
-}
-
-func (c *Client) doHttpRequestWithRetries(req *http.Request, retries int, retryBackoffDuration time.Duration) (*http.Response, error) {
-	resp, temporaryErr, err := c.doHttpRequest(req)
-	if temporaryErr && retries > 0 {
-		if err == nil {
-			_ = resp.Body.Close() // We want to retry anyway.
-		}
-		var timeToSleep time.Duration
-		if d, ok := parseRetryAfter(resp.Header.Get("Retry-After")); ok {
-			timeToSleep = d
-		} else {
-			timeToSleep = retryBackoffDuration
-		}
-		timer := time.NewTimer(timeToSleep)
-		defer timer.Stop()
-		select {
-		case <-timer.C:
-		case <-req.Context().Done():
-			return nil, req.Context().Err()
-		}
-
-		return c.doHttpRequestWithRetries(req, retries-1, retryBackoffDuration*2)
-	}
-
-	return resp, err
-}
-
-func parseRetryAfter(val string) (time.Duration, bool) {
+func parseRetryAfter(val string) time.Duration {
 	if n, err := strconv.ParseUint(val, 10, 64); err == nil {
-		return time.Duration(n) * time.Second, true
+		return time.Duration(n) * time.Second
 	}
 	if t, err := time.Parse(time.RFC1123, val); err == nil {
-		return time.Until(t), true
+		return time.Until(t)
 	}
-	return 0, false
+	return 0
 }
 
-func (c *Client) callAPI(ctx context.Context, method, path string, reqBody, resType interface{}, query ...string) error {
+func (c *Client) callAPIonce(ctx context.Context, method, path string, reqBody, resType interface{}, query ...string) (err error) {
+	traceData := map[string]interface{}{
+		"method": method,
+		"path":   path,
+	}
+
+	defer func() {
+		if err == nil {
+			tflog.Trace(ctx, "api call succeeded", traceData)
+			return
+		}
+		tflog.Trace(ctx, "api call failed:"+err.Error(), traceData)
```

Comment:
perhaps we can keep the success path to the left, so reverse the if stmt

Reply:
done